presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -171,6 +172,28 @@ default void renameColumn(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Get the physical layout for a inserting into an existing table.
+     */
+    default Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        List<ConnectorTableLayout> layouts = getTableLayouts(session, tableHandle, new Constraint<>(TupleDomain.all(), map -> true), Optional.empty())
+                .stream()
+                .map(ConnectorTableLayoutResult::getTableLayout)
+                .filter(layout -> layout.getNodePartitioning().isPresent())
+                .collect(toList());
+
+        if (layouts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (layouts.size() > 1) {
+            throw new PrestoException(NOT_SUPPORTED, "Tables with multiple layouts can not be written");
```

Comment:
The code doesn't match the error string. The code says: Table with multiple layouts can still be written to as long as no more than one of them has `nodePartitioning`. And I suspect the code is wrong.

* If there isn't a connector that depends on this behavior, I believe this should be fixed.
* If there is, that connector should implement override this method, because the connector can make this decision explicitly.

Regardless, this problem doesn't have to be fixed in this PR because it was like this before.